mlflow/utils/environment.py

```diff
@@ -9,15 +9,18 @@
 """
 
 
-def _mlflow_conda_env(path, additional_conda_deps=None, additional_pip_deps=None,
-                      additional_conda_channels=None):
+def _mlflow_conda_env(path=None, additional_conda_deps=None, additional_pip_deps=None,
+        additional_conda_channels=None):
     """
     Create conda environment file. Contains default dependency on current python version.
-    :param path: local filesystem path where the conda env file is to be created.
+    :param path: Local filesystem path where the conda env file is to be created. If unspecified,
+                 the conda env will be returned as a string.
     :param additional_conda_deps: List of additional conda dependencies passed as strings.
     :param additional_pip_deps: List of additional pip dependencies passed as strings.
     :param additional_channels: List of additional conda channels to search when resolving packages.
-    :return: path where the conda environment file has been created.
+    :return: Either:
```

Comment:
Can we avoid this non-uniformity of returned values. For instance, if the client has specified `path` -- does that call stack need to be returned that same string back?

Reply:
Done! We always return the environment text, regardless of whether or not the environment was written to the filesystem.